src/managers/GuildMemberManager.js

```diff
@@ -135,6 +136,48 @@ class GuildMemberManager extends BaseManager {
     return this._fetchMany(options);
   }
 
+  /**
+   * Edits a member of the guild.
+   * <warn>The user must be a member of the guild</warn>
+   * @param {UserResolvable} user The member to edit
+   * @param {GuildMemberEditData} data The data to edit the member with
+   * @param {string} [reason] Reason for editing this user
+   * @returns {Promise<void>}
+   */
+  async edit(user, data, reason) {
+    const id = this.client.users.resolveID(user);
+    if (!id) throw new TypeError('INVALID_TYPE', 'user', 'UserResolvable');
+
+    if (data.channel) {
```

Comment:
This should really not modify the input object as if I pass a frozen data object, it'll throw an error. You probably want to create an _data local obj and add props to it from the input

Reply:
Are you referring to the way `data` is modified by resolving variables? While I'd technically agree with the princples you're brining up here, that part of the code is directly lifted from the existing `GuildMember#edit` method. Modifying data objects is done pretty extensively through Discord.js, we don't really follow principle of not mutating data well at all.

Having said that, not really an excuse not to start! I can definitely incorporate this.

Reply:
We can slowly fix that and remove all mutations on the library, or we can do that when we switch to TypeScript.